framework/src/outputs/FileOutput.C

```diff
@@ -74,12 +77,24 @@ FileOutput::FileOutput(const InputParameters & parameters) :
     _file_base += buffer;
   }
 
-  // Check the file directory of file_base
+  // Check the file directory of file_base and create if needed
   std::string base = "./" + _file_base;
   base = base.substr(0, base.find_last_of('/'));
   if (access(base.c_str(), W_OK) == -1)
-    mooseError("Can not write to directory: " + base + " for file base: " + _file_base);
-
+  {
+    //Directory does not exist. Loop through incremental directories and create as needed.
+    size_t pos = 0;
+    std::string inc_path;
+    do
```

Comment:
An \/ would be part of a file or directory name. Tokenize would split the
path here.

On Fri, May 27, 2016, 5:52 PM Derek Gaston <notifications@github.com> wrote:

> In framework/src/outputs/FileOutput.C
> <https://github.com/idaholab/moose/pull/7101#discussion_r64979007>:
>
> >    std::string base = "./" + _file_base;
> >    base = base.substr(0, base.find_last_of('/'));
> >    if (access(base.c_str(), W_OK) == -1)
> > -    mooseError("Can not write to directory: " + base + " for file base: " + _file_base);
> > -
> > +  {
> > +    //Directory does not exist. Loop through incremental directories and create as needed.
> > +    size_t pos = 0;
> > +    std::string inc_path;
> > +    do
>
> Why doesn't it? I don't see anything in it that would preclude it dealing
> with backslash escapes...
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7101/files/bd2be34efac386b494aa5f1183bfd7561098d68d#r64979007>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAMWPpQHvI2Q0NxG2t_uvQKFHXPRrMghks5qF4PMgaJpZM4IoKUf>
> .
>